src/esp/physics/bullet/BulletArticulatedObject.h

```diff
@@ -0,0 +1,401 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_PHYSICS_BULLET_BULLETARTICULATEDOBJECT_H_
+#define ESP_PHYSICS_BULLET_BULLETARTICULATEDOBJECT_H_
+
+/** @file
+ * @brief Class @ref esp::physics::BulletArticulatedObject
+ */
+
+#include <utility>
+
+#include "../ArticulatedObject.h"
+#include "BulletBase.h"
+#include "BulletDynamics/Featherstone/btMultiBodyJointMotor.h"
+#include "BulletDynamics/Featherstone/btMultiBodySphericalJointMotor.h"
+#include "BulletURDFImporter.h"
+
+namespace esp {
+
+namespace physics {
+
+// forward delcaration from BulletURDFImporter
+struct JointLimitConstraintInfo;
+
+////////////////////////////////////
+// Link
+////////////////////////////////////
+
+class BulletArticulatedLink : public ArticulatedLink, public BulletBase {
+ public:
+  BulletArticulatedLink(scene::SceneNode* bodyNode,
+                        const assets::ResourceManager& resMgr,
+                        std::shared_ptr<btMultiBodyDynamicsWorld> bWorld,
+                        int index,
+                        std::shared_ptr<std::map<const btCollisionObject*, int>>
+                            collisionObjToObjIds)
+      : ArticulatedLink(bodyNode, index, resMgr),
+        BulletBase(std::move(bWorld), std::move(collisionObjToObjIds)) {}
+
+  Magnum::Range3D getCollisionShapeAabb() const override {
+    // TODO: collision object should be linked here
```

Comment:
Should we resolve this todo before merging this? Or at least print a louder error for users trying to use this non-implemented functionality?

Reply:
This is really part of a larger refactor to bring per-link functionality into the ArticulatedLink class instead of handling it at the ArticulatedObject level. I added an un-implemented warning for now.